ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V114__activate_string_interning_for_parties_and_templates.sql

```diff
@@ -0,0 +1,736 @@
+    exercise_child_event_ids TEXT[] NOT NULL,
+    -- * compression flags
+    create_key_value_compression SMALLINT,
+    exercise_argument_compression SMALLINT,
+    exercise_result_compression SMALLINT
+);
+
+SELECT 'DEBUG populate new table...';
+
+-- migrate data
+INSERT INTO participant_events_non_consuming_exercise_interned
+SELECT
+    event_sequential_id,
+    ledger_effective_time,
+    node_index,
+    event_offset,
+    transaction_id,
+    workflow_id,
+    command_id,
+    application_id,
+    CASE WHEN submitters IS NULL THEN NULL ELSE COALESCE(
+        (
+            SELECT array_agg(string_interning.internal_id)
+            FROM unnest(submitters) string_array(external_string)
+            INNER JOIN string_interning
+            ON (string_interning.external_string = 'p|' || string_array.external_string)
+        ),
+        ARRAY[]::INTEGER[]
+    ) END AS submitters,
+    event_id,
+    contract_id,
+    string_interning.internal_id AS template_id,
+    COALESCE(
+        (
+            SELECT array_agg(string_interning.internal_id)
+            FROM unnest(flat_event_witnesses) string_array(external_string)
+            INNER JOIN string_interning
+            ON (string_interning.external_string = 'p|' || string_array.external_string)
+        ),
+        ARRAY[]::INTEGER[]
+    ) AS flat_event_witnesses,
+    COALESCE(
+        (
+            SELECT array_agg(string_interning.internal_id)
+            FROM unnest(tree_event_witnesses) string_array(external_string)
+            INNER JOIN string_interning
+            ON (string_interning.external_string = 'p|' || string_array.external_string)
+        ),
+        ARRAY[]::INTEGER[]
+    ) AS tree_event_witnesses,
+    create_key_value,
+    exercise_choice,
+    exercise_argument,
+    exercise_result,
+    COALESCE(
+        (
+            SELECT array_agg(string_interning.internal_id)
+            FROM unnest(exercise_actors) string_array(external_string)
+            INNER JOIN string_interning
+            ON (string_interning.external_string = 'p|' || string_array.external_string)
+        ),
+        ARRAY[]::INTEGER[]
+    ) AS exercise_actors,
+    exercise_child_event_ids,
+    create_key_value_compression,
+    exercise_argument_compression,
+    exercise_result_compression
+FROM participant_events_non_consuming_exercise
+LEFT OUTER JOIN string_interning
+ON ('t|' || participant_events_non_consuming_exercise.template_id = string_interning.external_string);
+
+SELECT 'DEBUG add indexes...';
+
+-- rename old indexes
+ALTER INDEX participant_events_non_consuming_exercise_event_offset RENAME TO participant_events_non_consuming_exercise_event_offset_old;
+ALTER INDEX participant_events_non_consuming_exercise_event_sequential_id RENAME TO participant_events_non_consuming_exercise_event_sequential_id_old;
+ALTER INDEX participant_events_non_consuming_exercise_event_id_idx RENAME TO participant_events_non_consuming_exercise_event_id_idx_old;
```

Comment:
fyi @nmarton-da - the length of the rename target seems to be too long for postgres (63 char limit?):

```
WARN  o.f.c.i.s.DefaultSqlScriptExecutor - DB: identifier "participant_events_non_consuming_exercise_event_sequential_id_old" will be truncated to "participant_events_non_consuming_exercise_event_sequential_id_o" (SQL State: 42622 - Error Code: 0)
```

Seems more a fit & finish item. Not sure if we have time/priority to fix this.